crates/services/executor/src/executor.rs

```diff
@@ -159,18 +163,97 @@ impl TransactionsSource for OnceTransactionsSource {
     }
 }
 
-/// ! The executor is used for block production and validation. Given a block, it will execute all
-/// the transactions contained in the block and persist changes to the underlying database as needed.
-/// In production mode, block fields like transaction commitments are set based on the executed txs.
-/// In validation mode, the processed block commitments are compared with the proposed block.
+/// The executor is used for block production and validation.
+/// It acts like a factory creating an `ExecutionInstance` valid only for one block.
```

Comment:
I think this is an implementation detail. I don't know if it caller needs to know this.